framework/doc/content/source/meshgenerators/ParsedSubdomainIDsGenerator.md

```diff
@@ -0,0 +1,19 @@
+# ParsedSubdomainIDsGenerator
+
+!syntax description /Mesh/ParsedSubdomainIDsGenerator
+
+## Overview
+
+The `ParsedSubdomainIDsGenerator` generator allows the user to specify parsed expressions to compute the subdomain IDs of elements in the mesh.
+
+The parsed expression is provided by [!param](/Mesh/ParsedSubdomainIDsGenerator/expression), which can a function of the element centroid coordinates `x`, `y`, and `z`. More importantly, users can use the extra element integer values of each element as the function variables. The names of these extra element integers can be provide through [!param](/Mesh/ParsedSubdomainIDsGenerator/extra_element_id_names). This feature allows users to assign subdomain IDs based on the values of these extra element integers.
+
+When the subdomain IDs are reassigned by this generator, [!param](/Mesh/ParsedSubdomainIDsGenerator/excluded_subdomains) can be used to specify a list of subdomains that need to be retained.
+
+Note that due the flexibility provided by the parsed expression, the function value could be a non-integer value or negative. In that case, the absolute value of the rounded function value is used as the subdomain ID.
```

Comment:
I think I would rather an error for a negative value? (rounding is fine) What is your use case for a negative value ?